src/main/proto/TokenMint.proto

```diff
@@ -33,7 +33,16 @@ The operation increases the Total Supply of the Token. The maximum total supply
 The amount provided must be in the lowest denomination possible. Example:
 Token A has 2 decimals. In order to mint 100 tokens, one must provide amount of 10000. In order to mint 100.55 tokens, one must provide amount of 10055.
  */
+
+message AmountOrMemo {
+  oneof amountOrMemo {
+        int64 amount = 1; // Applicable to tokens of type FUNGIBLE. The amount to mint to the Treasury Account. Amount must be a positive non-zero number represented in the lowest denomination of the token. The new supply must be lower than 2^63
+        string memo = 2; // Applicable to tokens of type NON_FUNGIBLE. The metadata for the given NFT instance that is being created. Maximum allowed size is 100 bytes
+      }
+}
+
```

Comment:
With this implementation, a hacky ( in java) solution must be made - extracting amounr OR memo from amountOrMemo, with no methods for checking if any is present. However, we can use the google/protobuf/wrappers.proto wrappers, which provide us with optional values. 
Example:
```

message AmountOrMemo {
  oneof amountOrMemo {
    google.protobuf.Int64Value amount = 2;
    google.protobuf.StringValue memo = 3;
    }
}
```